src/coordinator/storage/local/storage.go

```diff
@@ -190,7 +329,119 @@ func newWriteRequest(writeRequestCommon *writeRequestCommon, timestamp time.Time
 	}
 }
 
-func (s *localStorage) Close() error {
-	s.namespace.Finalize()
-	return nil
+type multiFetchResult struct {
+	sync.Mutex
+	result           *storage.FetchResult
+	err              xerrors.MultiError
+	dedupeFirstAttrs storage.Attributes
+	dedupeMap        map[string]multiFetchResultSeries
+}
+
+type multiFetchResultSeries struct {
+	idx   int
+	attrs storage.Attributes
+}
+
+func (r *multiFetchResult) add(
+	attrs storage.Attributes,
+	result *storage.FetchResult,
+	err error,
+) {
+	r.Lock()
+	defer r.Unlock()
+
+	if err != nil {
+		r.err = r.err.Add(err)
+		return
+	}
+
+	if r.result == nil {
```

Comment:
is it a bug to initialize r.result but not r.dedupeMap ? Seems like this may not work since for the second result, you will just add duplicate things to the back of the seriesList

Reply:
yeah I missed the part of it creating the dedup map with previous result.